### ISSUE TITLE
Feature pipeline refitted on eval data in --data-path mode, policy receives OOD observations

### ISSUE THREAD

**Issue by kwojdalski** [kwojdalski/masters_thesis#90]
**File:** src/cli/commands/evaluate_command.py:294-296
**Category:** 3 — State Not Carried Across Boundaries
**Severity:** HIGH

**What it computes:** When `--data-path` is used, `_prepare_arbitrary_df` calls `pipeline.fit(df)` on the arbitrary eval data and then `pipeline.transform(df)` on the same data. Normalization statistics (mean, std) are derived from the eval file.

**What it should compute:** The feature pipeline should use normalization statistics from the training data — i.e., the same statistics that were in effect when the policy was trained. The trained policy learned to process observations normalized with training-data statistics; presenting it with observations normalized against eval-data statistics produces out-of-distribution (OOD) inputs.

**Root cause:**
```python
# evaluate_command.py lines 291-298
feature_config = getattr(config.data, "feature_config", None)
if feature_config:
    from trading_rl.features import FeaturePipeline
    pipeline = FeaturePipeline.from_yaml(feature_config)
    pipeline.fit(df)          # <-- fits on eval data, not training data
    features = pipeline.transform(df)
    df = pd.concat([df, features], axis=1)
```

The checkpoint format saves network weights and optimizer state but not the fitted scaler parameters (`RunningMeanStd.mean`, `RunningMeanStd.var`, `StandardScaler.mean_`, etc.). There is no mechanism to restore training-time normalization statistics.

**Proposed fix:**
Short-term: add a `logger.warning` making the user aware that normalization statistics are derived from the eval data, not from training, and that reported metrics may not reflect true out-of-sample performance.

Long-term: save the fitted scaler `state_dict` (already defined on `RunningMeanStd` and `TimeWeightedRunningMeanStd`) alongside the checkpoint, and reload it during standalone eval so that the same normalization is applied.
